### PR TITLE
[GRACE] feat(connector): T-kassa ClientSDKSessionToken + Authorize

### PR DESCRIPTION
**Pull request by shuklatushar226**
## Summary

Implement the **T-kassa** (T-Bank / T-Business e-acquiring **v2**, base URL `https://securepay.tinkoff.ru/v2/`) connector with two flows, both backed by `POST /v2/Init`:

- **Authorize** — creates the payment; maps `PaymentId` -> `connector_transaction_id`, `PaymentURL` -> `redirection_data` (`RedirectForm::Uri`), and the T-Bank payment status -> UCS attempt status.
- **ClientSDKSessionToken** (`ClientAuthenticationToken`) — opens an SDK payment session (`Device=SDK`) and returns `PaymentId` + `PaymentURL` as the client SDK session token via a new `ConnectorSpecificClientAuthenticationResponse::TKassa` arm.

Generated and validated by **GRACE** (automated connector integration pipeline).

## Authentication

T-Bank authenticates each request with an in-body **`Token`** = SHA-256 (lowercase hex) of the sorted root-level request values plus the terminal **Password** (`sha2` + `hex`). The `TerminalKey:Password` pair is provided through the `HeaderKey` `api_key` and split in the transformer. No credentials are hardcoded in source.

## Changes

- New connector: `connectors/t_kassa.rs` (`create_all_prerequisites!` for Authorize + ClientAuthenticationToken, two `macro_connector_implementation!` blocks, `ConnectorCommon`, error mapping) and `connectors/t_kassa/transformers.rs` (Init request/response, Token signature, status mapping, both flow `TryFrom`s).
- Registration: `connectors.rs`, `types.rs`, `default_implementations.rs`, `field-probe/auth.rs`, `domain_types/{connector_types,router_data,types}.rs`, `grpc-api-types/proto/payment.proto` (enum `T_KASSA`, `TKassaConfig`, `TKassaClientAuthenticationResponse`), and `config/{development,sandbox,production}.toml`.

## gRPC Test Results

**Status: FAILED — credentials (placeholder terminal rejected by T-Bank)**

The build passes cleanly and both flows reach the live `/v2/Init` endpoint. With placeholder credentials, T-Bank returns **HTTP 200 / ErrorCode `501` ("Неверные параметры")** — the request and SHA-256 `Token` are well-formed, but the dummy terminal is rejected. There is no `creds.json` entry for T-kassa in this environment, so a 2xx success status cannot be obtained without real sandbox credentials.

<details>
<summary>grpcurl — Authorize (credentials redacted)</summary>

```
grpcurl -plaintext \
  -H 'x-connector: t_kassa' \
  -H 'x-auth: header-key' \
  -H 'x-api-key: <REDACTED:TerminalKey:Password>' \
  -d '{
    "merchant_transaction_id": "test_tkassa_authorize_001",
    "amount": {"minor_amount": 19200, "currency": "RUB"},
    "payment_method": {"card": {"card_number": {"value": "<REDACTED>"}, "card_exp_month": {"value": "03"}, "card_exp_year": {"value": "2030"}, "card_cvc": {"value": "<REDACTED>"}, "card_holder_name": {"value": "John Doe"}}},
    "capture_method": "AUTOMATIC",
    "auth_type": "NO_THREE_DS",
    "return_url": "https://example.com/return",
    "description": "Probe payment",
    "address": {"billing_address": {"first_name": {"value": "John"}, "last_name": {"value": "Doe"}, "line1": {"value": "Tverskaya 1"}, "city": {"value": "Moscow"}, "zip_code": {"value": "<REDACTED>"}, "country_alpha2_code": "RU"}}
  }' \
  localhost:18000 types.PaymentService/Authorize

Response:
{
  "status": "FAILURE",
  "error": { "connectorDetails": { "code": "501", "message": "Неверные параметры.", "reason": "Неверные параметры." } },
  "statusCode": 200
}
```
</details>

<details>
<summary>grpcurl — ClientSDKSessionToken (credentials redacted)</summary>

```
grpcurl -plaintext \
  -H 'x-connector: t_kassa' \
  -H 'x-auth: header-key' \
  -H 'x-api-key: <REDACTED:TerminalKey:Password>' \
  -d '{
    "merchant_client_session_id": "test_tkassa_sdk_session_001",
    "payment": { "amount": {"minor_amount": 19200, "currency": "RUB"}, "return_url": "https://example.com/return" }
  }' \
  localhost:18000 types.MerchantAuthenticationService/CreateClientAuthenticationToken

Response:
{
  "error": { "connectorDetails": { "code": "501", "message": "Неверные параметры.", "reason": "Неверные параметры." } },
  "statusCode": 200
}
```
</details>

## Validation Checklist

- [x] `cargo build --package connector-integration` passes with zero errors/warnings
- [x] grpc-server builds; both flows route to `/v2/Init` and parse the connector response
- [ ] grpcurl returned a 2xx success status — blocked: no real T-kassa sandbox credentials (HTTP 200 / code 501)
- [x] No credentials in committed source code
- [x] Only connector + required registration/domain-type files modified

> Auto-generated by GRACE and marked **do not merge** because end-to-end validation could not reach a success status without real credentials. Provide a sandbox `TerminalKey:Password`, re-run the grpcurl tests, and remove the label once a 2xx success is confirmed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by XyneSpaces** [comment]
## Review Summary

**PR Type:** GRACE-generated connector (BLOCKED — DO NOT MERGE)  
**Status:** FAILED — Credentials rejected by T-Bank  
**Risk Level:** N/A — Cannot assess without successful test

---

### Analysis

**Failure Reason:**
Placeholder/dummy terminal credentials rejected by T-Bank with ErrorCode `501` ("Неверные параметры" — Invalid parameters). Request structure and SHA-256 Token computation are correct; the terminal simply doesn't exist in T-Bank's system.

**Code Quality (Static):**
- `cargo build --package connector-integration` passes
- Both flows route to `/v2/Init` correctly
- SHA-256 Token signature implementation follows T-Bank spec

**Technical Details:**
- Auth: `HeaderKey` with `TerminalKey:Password` in `api_key`
- Request correctly sorted for Token computation
- Both Authorize and ClientSDKSessionToken reach live endpoint

### Blocking Issues

:rotating_light: **DO NOT MERGE — No successful test**

While the HTTP 200 / ErrorCode 501 confirms the request is well-formed, a SUCCESS (2xx + success status) is required per GRACE workflow. The placeholder rejection demonstrates:
1. ✓ Base URL correct (`https://securepay.tinkoff.ru/v2`)
2. ✓ Token signature computation correct (gets past signature validation)
3. ✗ Terminal credentials invalid (expected with placeholders)

### Required Actions

1. Obtain real T-Bank sandbox `TerminalKey:Password`
2. Re-run grpcurl tests for both flows
3. Verify success response contains `PaymentId` and `PaymentURL`
4. Confirm status mapping works for `NEW`, `CONFIRMED`, `REJECTED` states
5. Remove "do not merge" label once 2xx success confirmed

File: crates/integrations/connector-integration/src/connectors/t_kassa/transformers.rs [hunk 1]
```diff
@@ -0,0 +1,472 @@
+use common_enums::AttemptStatus;
```
Comment (XyneSpaces) [inline crates/integrations/connector-integration/src/connectors/t_kassa/transformers.rs:1]: :warning: Token signature cannot be fully validated — placeholder credentials return ErrorCode `501` (invalid parameters), meaning the SHA-256 `Token` construction is syntactically correct but the dummy terminal is rejected.

Once real sandbox credentials are available, verify the signature matches T-Bank's expected format: SHA-256 of sorted request values + Password (lowercase hex).

File: crates/integrations/connector-integration/src/connectors/t_kassa/transformers.rs [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+use domain_types::{
```
Comment (XyneSpaces) [inline crates/integrations/connector-integration/src/connectors/t_kassa/transformers.rs:2]: :bulb: Consider adding handling for additional T-Kassa terminal states like `REJECTED`, `DEADLINE_EXPIRED`, or `REFUNDED` in the status mapping if they can appear in the `/v2/Init` response.

Document which statuses are expected from Init vs which appear only in callback notifications.

**Review by XyneSpaces — COMMENTED** [review]
## Review Summary

**Classification:** connector-new-integration · metadata: grace-generated-pr  
**Risk:** Medium  
**Verdict:** :arrows_counterclockwise: Request Changes

:warning: 2 warnings

**Status:** PR marked `[GRACE]` and labeled `do not merge` due to test credential issues (placeholder terminal rejected by T-Bank with ErrorCode `501`). Code structure is sound but validation cannot complete without real sandbox credentials.

---

### Warnings

**1. Cannot validate SHA-256 Token signature correctness**
The request reaches T-Bank's `/v2/Init` endpoint and returns HTTP 200 with ErrorCode `501`, indicating the request format and SHA-256 Token signature are syntactically correct. However, without real credentials, the signature cannot be validated against T-Bank's actual verification logic.

**2. Incomplete enum mapping for T-Kassa status values**
The `TransactionStatus` enum mapping should verify all possible T-Bank response statuses are handled, particularly edge cases like `REJECTED` or `DEADLINE_EXPIRED`.

---

**Action required:** Obtain valid T-Kassa sandbox `TerminalKey:Password` credentials, re-run grpcurl tests to confirm 2xx success, then remove `do not merge` label.
